homework/tests/example.spec.js

```diff
@@ -17,15 +20,546 @@ const data = [
 
 data.forEach(version => {
   test.describe(version + ': Add', () => {
-    test('Concatenating 2 and 3 results in 23', async ({ page }) => {
-      await page.goto('https://testsheepnz.github.io/BasicCalculator');
-      await page.selectOption('#selectBuild', { label: version});
-      await page.locator('#number1Field').type('2');
-      await page.locator('#number2Field').type('3');
-      await page.selectOption('#selectOperationDropdown', {label: 'Add'});
-      await page.locator('#calculateButton').click();
-  
-      await expect(page.locator('#numberAnswerField')).toHaveValue('5');
+    test('Adding 2 and 3 results in 5', async ({ page }) => {
+      let searchPage = new SearchPage(page);
+      await searchPage.navigate();
+      await searchPage.selectBuild(version);
+
+      let calcForm = new CalcForm(page);
+      await calcForm.enterFirstNumber('2');
+      await calcForm.enterSecondNumber('3');
+      await calcForm.selectOperation('Add');
```

Comment:
you have grouped tests in a way that within one describe block, all the tests have the same operation.  Is there a reason you keep writing it every time, instead of having a variable that could be used in every test?